### PR TITLE
MEAL-47 recipe DTO

### PR DESCRIPTION
**Pull request by seanyles**

### CODE REVIEW

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.app.eece4792mealmaster.dto;
+
+public class RecipeIngredientDto {
+    private Long ingredient;
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java]: ingredient corresponds to the GF id, correct?

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    private Double servings;
+
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java]: does this class no longer need an id?

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientId.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package com.app.eece4792mealmaster.dto;
+
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Embeddable
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientId.java]: Where is this class being used?

File: src/main/java/com/app/eece4792mealmaster/services/RecipeService.java [hunk 4]
```diff
@@ -1,16 +1,24 @@
...
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
+import java.text.ParseException;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 import java.util.Optional;
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/services/RecipeService.java]: will have merge conflicts

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java (hunk 2 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java]: it's a dto, so no. This is strictly what the user interfaces with in the JSON body of the API

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientId.java (hunk 3 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientId.java]: embedded in RecipeINgredient

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java (hunk 1 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java]: yes

File: src/main/java/com/app/eece4792mealmaster/services/RecipeService.java (hunk 4 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/services/RecipeService.java]: cool beans

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java (hunk 1 above)
Comment (otuffaha) [reply src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java]: do you think it is worth noting that somehow?

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientId.java (hunk 3 above)
Comment (otuffaha) [reply src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientId.java]: i see the import, but is the logic missing in that class? or is there something going on behind the scenes?

Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientId.java]: it's a field that gets set in the constructor

File: src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java (hunk 1 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/dto/RecipeIngredientDto.java]: meh

**Pull request merged**